pkg/status/status.go

```diff
@@ -152,16 +158,23 @@ func GetDCAStatus() (map[string]interface{}, error) {
 	apiCl, err := apiserver.GetAPIClient()
 	if err != nil {
 		stats["custommetrics"] = map[string]string{"Error": err.Error()}
-		return stats, nil
+	} else {
+		stats["custommetrics"] = custommetrics.GetStatus(apiCl.Cl)
+	}
+
+	for _, c := range callbacks {
+		name, value := c()
```

Comment:
I get the idea but this looks a bit like premature optimization to me, because the slice of callbacks only contains one function.
Also I had some troubles following the code flow when reviewing, which hints that the code could potentially be difficult to maintain because you have to look at which callbacks are injected when reading the function.
IMHO I'd prefer to have some code duplication and an easier flow with a single `if` statement injecting the cluster checks status for now. WDYT?
Do you have any other ideas on how to make this simpler?

Reply:
Removed the callback logic and exposed a package-global `clusterchecks.GetStats()` function for use by the status cmd. WDYT?